collection_checklist.md

```diff
@@ -28,6 +28,8 @@ Every comment should say whether the reviewer expects it to be addressed, or whe
 - [ ] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
 - [ ] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
 - [ ] has changelog, preferably with `changelogs/changelog.yaml`
+- [ ] collection Galaxy artifact (tarball) should not contain any large objects (binaries) comparatively to the current tarball size limit of 20 MB
```

Comment:
```suggestion
- [ ] collection Galaxy artifact (tarball) should not contain any large objects (binaries) comparatively to the current tarball size limit of 20 MB
```
i would also mention other unrelated things like David mentioned (iirc pyenv files, etc.)

Reply:
This is complicated due to legal requirements. The tarballs on Galaxy act right now as the source distribution (and in fact only distribution), so it's not clear to me whether it would be allowed to leave anything out that's used to develop and package the collection.

Reply:
I think I'll create a new PR instead anyway as it would be harder to rebase this stuff. I'd like mentioning that we should check for such things anyway and point out to them. If a collection repo is full of such things, it's not good